js/ui/lookup.js

```diff
@@ -489,36 +489,65 @@ const Lookup = DropDownList.inherit({
         }
     },
 
-    _setPopupPosition: function() {
-        if(!this.option('itemCenteringEnabled')) return;
+    _getDifferenceOffsets: function(selectedListItem) {
+        return selectedListItem.offset().top - $(this.element()).offset().top;
+    },
 
+    _getPopupOffset: function(selectedIndex) {
+        const popupHeight = this.option('dropDownOptions.height')();
+        const selectedListItem = $(this._list.element()).find('.' + LIST_ITEM_SELECTED_CLASS);
+        const differenceOfHeights = (selectedListItem.height() - $(this.element()).height()) / 2;
+        const listItemsCount = this._list.option('items').length;
+        const lookupOffset = $(this._list.element()).offset().top;
 
-        const selectedIndex = this._list.option('selectedIndex');
-        const flipped = this._popup._$wrapper.hasClass(LOOKUP_POPOVER_FLIP_VERTICAL_CLASS);
-        if(selectedIndex === -1 || flipped) return;
+        let offset = 0;
 
-        const selectedListItem = $(this._list.element()).find('.' + LIST_ITEM_SELECTED_CLASS);
+        if(1 < selectedIndex && selectedIndex < (listItemsCount - 2)) {
```

Comment:
I think we need to add some info about what this condition means. It is unclear now. Is it two 'border' items at the start and the end of the list?

Reply:
Yes, it is two items at the start and the end of the list, which cannot be centered. What do you suggest? Add comment?

Reply:
I prefer the self-documenting code. We can extract this to the function for example

Reply:
We can extract internal code from 'If' condition, but it will not more clear

Reply:
it depends on function name :)

Reply:
What do you think about 'getCenteringOffset' name?

Reply:
needCentering(){ return selectedIndex > 1 && selectedIndex < (listItemsCount - 2);}
smth like this?

Reply:
Updated